test_brain_friendly.py

```diff
@@ -186,63 +187,43 @@ def test_invalid_braces():
-               '>-----.>->+++..+++.>-.<<+[>[+>+]>>]<--------------.>>.+++.'
-               '------.--------.>+.>+.')
-    output = io.BytesIO()
-    eval_program(program, 0, [0]*100, output=output)
-    assert output.getvalue() == b'Hello World!\n'
+    prefix = 'test_programs/hello_world/hello_world'
+    suffixes = ['_commented.bf', '_no_comma.bf',
+    '_no_comma2.bf', '_lowercase.bf', '_no_newline.bf']
+    filenames = [prefix+suffix for suffix in suffixes]
+    # Not all of these are the same.
+    file_outputs = [b'Hello World!\n', b'Hello World!\n',
+                    b'Hello World!\n', b'hello world', b'Hello, World!']
+
+    for (filename, file_output) in zip(filenames, file_outputs):
+        output = io.BytesIO()
+        eval_file(filename, 0, [0]*100, output=output)
+        output.seek(0)
+        assert output.getvalue() == file_output
 
+    # Visit the following link to find items that will fail non-wrapping, etc.
     # http://codegolf.stackexchange.com/questions/55422/hello-world/68494#68494
-    program = ('--->->->>+>+>>+[++++[>+++[>++++>-->+++<<<-]<-]<+++]'
-               '>>>.>-->-.>..+>++++>+++.+>-->[>-.<<]')
-    output = io.BytesIO()
-    eval_program(program, 0, [0]*100, output=output)
-    assert output.getvalue() == b'Hello, World!'
-
-    program = '''\
-+++++ +++               Set Cell #0 to 8
-[
-    >++++               Add 4 to Cell #1; this will always set Cell #1 to 4
-    [                   as the cell will be cleared by the loop
-        >++             Add 4*2 to Cell #2
-        >+++            Add 4*3 to Cell #3
-        >+++            Add 4*3 to Cell #4
-        >+              Add 4 to Cell #5
-        <<<<-           Decrement the loop counter in Cell #1
-    ]                   Loop till Cell #1 is zero
-    >+                  Add 1 to Cell #2
-    >+                  Add 1 to Cell #3
-    >-                  Subtract 1 from Cell #4
-    >>+                 Add 1 to Cell #6
-    [<]                 Move back to the first zero cell you find; this will
-                        be Cell #1 which was cleared by the previous loop
-    <-                  Decrement the loop Counter in Cell #0
-]                       Loop till Cell #0 is zero
-
-The result of this is:
-Cell No :   0   1   2   3   4   5   6
-Contents:   0   0  72 104  88  32   8
-Pointer :   ^
-
->>.                     Cell #2 has value 72 which is 'H'
->---.                   Subtract 3 from Cell #3 to get 101 which is 'e'
-+++++ ++..+++.          Likewise for 'llo' from Cell #3
->>.                     Cell #5 is 32 for the space
-<-.                     Subtract 1 from Cell #4 for 87 to give a 'W'
-<.                      Cell #3 was set to 'o' from the end of 'Hello'
-+++.----- -.----- ---.  Cell #3 for 'rl' and 'd'
->>+.                    Add 1 to Cell #5 gives us an exclamation point
->++.                    And finally a newline from Cell #6'''
+
+
+# Test reading from a file
```

Comment:
We should probably put these tests above the hello_world tests, since those rely on file IO.